### ISSUE TITLE
SC2048 reported inside [[ ... ]]

### ISSUE THREAD

**Issue by mkoskar** [koalaman/shellcheck#976]
#### For bugs
- Rule Id: `SC2048`
- My shellcheck version: `version: 0.4.6`
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
[[ $* =~ pattern ]]
```

#### Here's what shellcheck currently says:
SC2048 reported

#### Here's what I wanted or expected to see:
SC2048 **not** reported

**Comment by koalaman** [comment]
Nice catch! This old test had never been updated to take advantage of the later, more accurate "when-are-quotes-needed" detection.
